omegaconf/listconfig.py

```diff
@@ -256,7 +255,7 @@ def append(self, item: Any) -> None:
                 ref_type=self.__dict__["_metadata"].element_type,
                 key=index,
                 value=item,
-                is_optional=_is_optional(self),
+                is_optional=item._is_optional() if isinstance(item, Node) else True,
                 parent=self,
```

Comment:
Prior to #696 we had `is_optional=OmegaConf.is_optional(item)`.
Here we use `is_optional=item._is_optional() if isinstance(item, Node) else True`, since `OmegaConf.is_optional` issues a deprecation warning.